### ISSUE TITLE
False Report of RCS1163

### ISSUE THREAD

**Issue by wmjordan** [JosefPihrt/Roslynator#1082]
**Product and Version Used**: 
Release on 2023-4-28.

**Steps to Reproduce**:
The following code triggers this issue to report RCS1163 on parameter `x` in class `D` and parameter `y` in class `E`.

````C#
internal class C
{
	public void RegisterRun(bool useD, int a, int b) {
		if (useD) {
			Run(new D().Run, a, b);
		}
		else {
			Run(new E().Run, a, b);
		}
	}

	static void Run(Action<int, int> action, int a, int b) {
		action(a, b);
	}

	class D
	{
		public void Run(int x, int y) {
			Console.WriteLine(y);
		}
	}

	class E
	{
		public void Run(int x, int y) {
			Console.WriteLine(x);
		}
	}
}
````

**Expected Behavior**:
The aforementioned parameters are necessary since they must match the signature of delegate `Action<,>`

**Comment by JosefPihrt** [comment]
This analyzer checks if a parameter is used in a method's body. In that sense, the diagnostic is correct (IDE0060 will be reported as well).

Unfortunately, the analyzer cannot detect if the method is used as a method group anywhere in the code. So if your case is a real case and not just "artificial" sample, then the correct approach is to suppress the diagnostic.
